apps/application-system/api/src/app/modules/application/files/file.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common'
+import { generateResidenceChangePdf } from './utils/pdf'
+import * as AWS from 'aws-sdk'
+import { uuid } from 'uuidv4'
+import { PDF_TYPES } from '@island.is/application/api-template-utils'
+import { Application } from './../application.model'
+import { FormValue } from '@island.is/application/core'
+import {
+  ParentResidenceChange,
+  PersonResidenceChange,
+} from '@island.is/application/templates/children-residence-change'
+import { NationalRegistryUser } from '@island.is/api/schema'
+
+@Injectable()
+export class FileService {
+  s3: AWS.S3
+  private one_minute = 60
+
+  constructor() {
+    this.s3 = new AWS.S3()
+  }
+
+  async createPdf(application: Application, type: PDF_TYPES): Promise<string> {
+    const answers = application.answers as FormValue
+    const externalData = application.externalData as FormValue
+
+    switch (type) {
+      case PDF_TYPES.CHILDREN_RESIDENCE_CHANGE: {
+        return await this.createResidenceChangePdf(answers, externalData)
+      }
+    }
+  }
+
+  private async createResidenceChangePdf(
+    answers: FormValue,
+    externalData: FormValue,
+  ): Promise<string> {
+    const parentBNationalRegistry = externalData.parentNationalRegistry as FormValue
+    const nationalRegistry = externalData.nationalRegistry as FormValue
+    const nationalRegistryData = nationalRegistry.data as NationalRegistryUser
+    const childrenAppliedFor = (answers.selectChild as unknown) as Array<
+      PersonResidenceChange
+    >
+    const parentB = (parentBNationalRegistry.data as unknown) as ParentResidenceChange
+
+    parentB.email = answers.parentBEmail as string
+    parentB.phoneNumber = answers.parentBPhoneNumber as string
+
+    const parentA: ParentResidenceChange = {
+      id: nationalRegistryData.nationalId,
+      name: nationalRegistryData.fullName,
+      ssn: nationalRegistryData.nationalId,
+      phoneNumber: answers.phoneNumber as string,
+      email: answers.email as string,
+      address: nationalRegistryData.address?.streetAddress as string,
+      postalCode: nationalRegistryData.address?.postalCode as string,
+      city: nationalRegistryData.address?.city as string,
+    }
+
+    const expiry = answers.expiry as string
+
+    const pdfBuffer = await generateResidenceChangePdf(
+      childrenAppliedFor,
+      parentA,
+      parentB,
+      expiry,
+    )
+
+    const id = uuid()
+    const fileName = `${parentA.ssn}/${id}.pdf`
+    const bucket = `${process.env.NODE_ENV}-legal-residence-change`
```

Comment:
Can we pull this in as an environment variable/secret instead of hardcoding it? We'd be dead in the water if `production-legal-residence-change` has been taken by someone else :)

Reply:
Makes sense. 😅  

First time adding a secret for this project. Could you tell me about that process? 

I get that they are stored in AWS Parameter store and need to be labeled with `dev` and can of course follow examples in the codebase but think it would be good to hear about it instead of guessing 🙏

Reply:
There's really nothing secret about bucket names, so no need for parameter store IMO. You can just add 3 different environment values to the `env` section in helm: https://github.com/island-is/helm/blob/main/src/services/application-system/application-system-api.ts#L11. Something like

```
FILE_SERVICE_PRESIGN_BUCKET: {
    dev: 'island-is-development-fs-presign-bucket',
    staging: 'island-is-staging-fs-presign-bucket',
    prod: 'island-is-production-fs-presign-bucket',
}
```

Then you'll get the environment variable `node.env.FILE_SERVICE_PRESIGN_BUCKET` in you pod in k8s.

Then ping one of the devops folks and we'll create the buckets for you :smiley:. We'll want some characteristics for the bucket, such as if we can delete files after some time, if we need versioned objects etc.